qstash/workflow/basics/context.mdx

```diff
@@ -191,6 +197,64 @@ type ResultType = {
 const result = await context.call<ResultType>( ... ); 
 ```
 
+By default, `context.call` only retries once if its request to the given URL fails.
+
+### context.waitForEvent
+
+Stops the workflow run until it is notified externally to continue.
+There is also a timeout setting which makes the workflow continue if it's not notified within the time frame.
+
+```javascript
+import { serve } from "@upstash/workflow/nextjs"
+
+export const { POST } = serve<{ topic: string }>(async (context) => {
+  const request = context.requestPayload
+
+  const {
+    eventData, // data passed in notify
+    timeout    // boolean denoting whether the step was notified or timed out
+  } = await context.waitForEvent(
+    "wait for some event",
+    "my-event-id",
+    1000 // 1000 second timeout
+  );
+})
+```
+
+A workflow run waiting for event can be notified in two ways:
+- `context.notify`: [Notify step explained below](/qstash/workflow/basics/context#context-notify)
+- `client.notify`: [Notify method of the Workflow Client](/qstash/workflow/basics/client#notify-waiting-workflow).
+
+### context.notify
+
+Stops the workflow run until it is notified externally to continue.
+There is also a timeout setting which makes the workflow continue if it's not notified within the time frame.
+
+```javascript
+import { serve } from "@upstash/workflow/nextjs"
+
+export const { POST } = serve<{ topic: string }>(async (context) => {
+  const payload = context.requestPayload
+
+  const {
+    notifyResponse // result of notify, which is a list of notified waiters
+  } = await context.notify("notify step", "my-event-Id", payload);
+})
+```
+
+`notifyResponse` is a list of `NotifyResponse` objects:
+
+```typescript
+export type NotifyResponse = {
+  waiter: Waiter;
+  messageId: string;
+  error: string;
+};
+```
+
+To learn more about the `Waiter` object, see [the client notify method](/qstash/workflow/basics/client#get-waiters-of-event).
```

Comment:
I think we can add a collapsible here that explains the Waiter type, at least the field names if not the explanations